### ISSUE TITLE
Error: Provider produced inconsistent result after apply

### ISSUE THREAD

**Issue by maxbischoff** [stackitcloud/terraform-provider-stackit#377]
After upgrading the provider to version `0.18.0`, all applies to `stackit_ske_cluster` resources cause a `Provider produced inconsistent result after apply` error.

I can cleanly reproduce the issue with following config:

```tf
terraform {
  required_version = "1.8.3"
  required_providers {
    stackit = {
      source  = "stackitcloud/stackit"
      version = "0.18.0"
    }
  }
}

provider "stackit" {
  region                          = "eu01"
  ske_custom_endpoint             = "https://ske.api.eu01.stg.stackit.cloud"
}

resource "stackit_ske_cluster" "api_cluster" {
  name               = "repro"
  project_id         = "<redacted>"

  extensions = {
    argus = {
      enabled = false
    }
  }

  node_pools = [{
    name         = "nodepool"
    machine_type = "m1.3"
    max_surge = "1"
    max_unavailable = "0"
    maximum         = "3"
    minimum         = "1"
    availability_zones = ["eu01-1"]
  }]
}
```

Output of `terraform apply`:

```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following
symbols:
  + create

Terraform will perform the following actions:

  # stackit_ske_cluster.api_cluster will be created
  + resource "stackit_ske_cluster" "api_cluster" {
      + extensions              = {
          + argus = {
              + enabled = false
            }
        }
      + id                      = (known after apply)
      + kube_config             = (sensitive value)
      + kubernetes_version_used = (known after apply)
      + maintenance             = (known after apply)
      + name                    = "repro"
      + node_pools              = [
          + {
              + availability_zones = [
                  + "eu01-1",
                ]
              + cri                = "containerd"
              + labels             = (known after apply)
              + machine_type       = "m1.3"
              + max_surge          = 1
              + max_unavailable    = 0
              + maximum            = 3
              + minimum            = 1
              + name               = "nodepool"
              + os_name            = "flatcar"
              + os_version_used    = (known after apply)
              + volume_size        = 20
              + volume_type        = "storage_premium_perf1"
            },
        ]
      + project_id              = "<redacted>"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

stackit_ske_cluster.api_cluster: Creating...
stackit_ske_cluster.api_cluster: Still creating... [10s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [20s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [30s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [40s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [50s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [1m0s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [1m10s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [1m20s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [1m30s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [1m40s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [1m50s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [2m0s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [2m10s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [2m20s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [2m30s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [2m40s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [2m50s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [3m0s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [3m10s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [3m20s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [3m30s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [3m40s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [3m50s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [4m0s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [4m10s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [4m20s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [4m30s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [4m40s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [4m50s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [5m0s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [5m10s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [5m20s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [5m30s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [5m40s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [5m50s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [6m0s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [6m10s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [6m20s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [6m30s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [6m40s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [6m50s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [7m0s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [7m10s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [7m20s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [7m30s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [7m40s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [7m50s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [8m0s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [8m10s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [8m20s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [8m30s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [8m40s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [8m50s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [9m0s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [9m10s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [9m20s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [9m30s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [9m40s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [9m50s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [10m0s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [10m10s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [10m20s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [10m30s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [10m40s elapsed]
stackit_ske_cluster.api_cluster: Still creating... [10m50s elapsed]
╷
│ Warning: The kubelogin field is set to null
│
│   with stackit_ske_cluster.api_cluster,
│   on main.tf line 16, in resource "stackit_ske_cluster" "api_cluster":
│   16: resource "stackit_ske_cluster" "api_cluster" {
│
│ Failed to get static token kubeconfig, which means the new credentials rotation flow might already been triggered for this cluster. If
│ you are already using the stackit_ske_kubeconfig resource you can ignore this warning. If not, you must use it to access this cluster's
│ short-lived admin kubeconfig.
╵
╷
│ Error: Provider produced inconsistent result after apply
│
│ When applying changes to stackit_ske_cluster.api_cluster, provider "provider[\"registry.terraform.io/stackitcloud/stackit\"]" produced
│ an unexpected new value: .extensions: was
│ cty.ObjectVal(map[string]cty.Value{"acl":cty.NullVal(cty.Object(map[string]cty.Type{"allowed_cidrs":cty.List(cty.String),
│ "enabled":cty.Bool})), "argus":cty.ObjectVal(map[string]cty.Value{"argus_instance_id":cty.NullVal(cty.String), "enabled":cty.False})}),
│ but now null.
│
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
╷
│ Error: Provider produced inconsistent result after apply
│
│ When applying changes to stackit_ske_cluster.api_cluster, provider "provider[\"registry.terraform.io/stackitcloud/stackit\"]" produced
│ an unexpected new value: .network: was null, but now cty.ObjectVal(map[string]cty.Value{"id":cty.NullVal(cty.String)}).
│
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
```

**Comment by vicentepinto98** [comment]
Hi @maxbischoff, this is a bug that is only occurring on QA and is related to a change on the SKE API. The SKE team is working on a fix. 

It is not related to the new provider version v0.18.0, it impacts all releases.

In the meantime, you can use the PROD stage if you want.

**Comment by maxbischoff** [comment]
I just re-tested on prod and also get the error there. That is also what I saw this morning in our prod clusters after updating.

For reference, this is my adjusted terraform config:

```
terraform {
  required_version = "1.8.3"
  required_providers {
    stackit = {
      source  = "stackitcloud/stackit"
      version = "0.18.0"
    }
  }
}

provider "stackit" {
  region                          = "eu01"
}

resource "stackit_ske_cluster" "api_cluster" {
  name               = "repro"
  project_id         = "<redacted>"

  extensions = {
    argus = {
      enabled = false
    }
  }

  node_pools = [{
    name         = "nodepool"
    machine_type = "m1.3"
    max_surge = "1"
    max_unavailable = "0"
    maximum         = "3"
    minimum         = "1"
    availability_zones = ["eu01-1"]
  }]

}
```

And I'm getting this error:

```
│ Error: Provider produced inconsistent result after apply
│
│ When applying changes to stackit_ske_cluster.api_cluster, provider "provider[\"registry.terraform.io/stackitcloud/stackit\"]" produced
│ an unexpected new value: .network: was null, but now cty.ObjectVal(map[string]cty.Value{"id":cty.NullVal(cty.String)}).
│
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
```

**Comment by vicentepinto98** [comment]
That is a different error related to the `network` field that was already fixed in the latest version [v0.18.1](https://github.com/stackitcloud/terraform-provider-stackit/releases/tag/v0.18.1)

**Comment by maxbischoff** [comment]
Ah, my initial report had both issues, I only looked at the `network`-related one I also saw on prod, though. Thanks for pointing that out.
